packages/pros/src/adi/gyro.rs

```diff
@@ -0,0 +1,59 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use core::ffi::c_double;
+
+use pros_sys::PROS_ERR;
+
+use crate::error::bail_on;
+
+pub struct AdiGyro {
+    port: u8,
+    reference: i32
```

Comment:
Naming: `reference` -> `raw`. Also: probably should use `pros_sys::adi::adi_gyro_t` for clarity, even though its just a type alias for i32.